backends/redis.go

```diff
@@ -55,99 +53,80 @@ func (e KeyError) Error() string {
 	return "Key not found"
 }
 
-func (r *RedisBackend) newRedisClusterPool() redis.UniversalClient {
+func (r *RedisBackend) Connect() (temporal.KeyValue, error) {
```

Comment:
We're not using the `temporal.KeyValue` return anywhere other than unit tests. What do you think about just returning an error?